### PR TITLE
merged with master and enabled plotting of strongermax/polymax

### PR DESCRIPTION
**Pull request by Hrancheng**
Extended plotting of input/output statistics to strongermax and polymax, used similarly to consmax by specifying the stats to be visualized. Command line would look like: 
python3 train.py --out_dir=out --device=cpu --eval_interval=2 --log_interval=1 --block_size=2 --batch_size=5 --n_layer=3 --n_head=4 --n_embd=16 --lr_decay_iters=2 --dtype="float32" --max_iter=20 --softmax_variant_attn="polymax" --statistic=output_mean

### CODE REVIEW

File: explorations/strongermax_sweep.json
Comment (gkielian) [inline explorations/strongermax_sweep.json:1]: Two large change diffs from the main branch, we should all sync on these two:

n_kv_group - this is the group query attention setting, might be good to set this to 6 for now to match the head size as we begin exploring.

max_iters - pretty sure that 8000 will converge/overfit which is what we generally want for the sweep.  Some settings might delay the min validation loss past 3500, so changing this to 8000 is probably a good idea for preliminary sweeps.

File: variations/softmax_variations.py
Comment (gkielian) [inline variations/softmax_variations.py:1]: We may have to do one last review pass after merging to ensure the the softmax_variations.py have logging features and latest dimensional features.

**Review by gkielian — COMMENTED** [review]
looks good
